google/cloud/internal/opentelemetry.h

```diff
@@ -26,6 +30,12 @@ namespace internal {
 
 bool TracingEnabled(Options const& options);
 
+opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> GetTracer(
+    Options const& options);
+
+opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
```

Comment:
Forgive my continuing naivety, but is a span something that should always (usually?) be stack based?

Reply:
The API to make a span returns it in a smart pointer: https://github.com/open-telemetry/opentelemetry-cpp/blob/8a9d085f05b04a2585390c1398f76c388cbcff0e/api/include/opentelemetry/trace/tracer.h#L40

So I think we should just pass that pointer around.